addons/website/models/website.py

```diff
@@ -1068,22 +1068,13 @@ def viewref(self, view_id, raise_if_not_found=True):
             raise ValueError('No record found for unique ID %s. It may have been deleted.' % (view_id))
         return view
 
-    @tools.ormcache_context(keys=('website_id',))
-    def _cache_customize_show_views(self):
-        views = self.env['ir.ui.view'].with_context(active_test=False).sudo().search([('customize_show', '=', True)])
-        views = views.filter_duplicate()
-        return {v.key: v.active for v in views}
-
     @tools.ormcache_context('key', keys=('website_id',))
-    def is_view_active(self, key, raise_if_not_found=False):
+    def is_view_active(self, key):
         """
-            Return True if active, False if not active, None if not found or not a customize_show view
+            Return True if active, False if not active, None if not found
         """
-        views = self._cache_customize_show_views()
-        view = key in views and views[key]
-        if view is None and raise_if_not_found:
-            raise ValueError('No view of type customize_show found for key %s' % key)
-        return view
+        view = self.viewref(key, raise_if_not_found=False)
```

Comment:
Well if there is no more auto discovery of customize show views and they all need to be explicitly added in the right panel through XML, then we:
1. Won't have a direct link between customize view and the right panel, this would be bug prone then, as one could either:
  A. Omit the customize_show property on the view, then its value would not be returned in the cached method, it would always be falsy
  B. Add the customize_show and omit the XML part, then it would not be toggable.
2. Don't really need to keep the customize_show field as it would only be used for the cache method

So this field should definitely be removed, and https://github.com/odoo/odoo/commit/697f6007847e0cf209610acd533c5cd4d2c83296 simply reverted as I told a few weeks ago.
For the record, it was following https://github.com/odoo/odoo/commit/54048cb2280ff3036a11f1b25bb36d25020b4887 which introduced the cache system for the customize views.
Reverting the mentioned commit would be a perf regression but not critical so I guess it is fine. It would only impact perfs in cold state.

> should not we let this crash instead of not raising the error?

It was chosen not to, and I'd keep it that way, as without this your DB is completely broken if the user somehow messed up with his DB and removed that view. Sometimes people just remove that view if they don't want that part, who knows why